### PR TITLE
Clean up subject selection

### PR DESCRIPTION
**Pull request by marten**
I've removed all of the checks to see if we need to call out to Cellect/CellectEx before enqueueing the jobs. Instead, it'll just do that when it runs. However, I can see a case to be made for keeping it. If you like, I would suggest doing that in an override to `perform_async` in the worker, rather than littering the rest of the codebase with it.

### CODE REVIEW

File: app/models/subject_selector.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class SubjectSelector
+  def self.for(workflow)
+    case strategy_for(workflow)
```
Comment (camallen) [inline app/models/subject_selector.rb]: could we push responsibility to the workflow to hide the implementation? perhaps
``` ruby
case 
when workflow.strategy?(:cellect)
...
```

File: app/workers/notify_subject_selector_of_seen_subject_worker.rb [hunk 2]
```diff
@@ -1,6 +1,4 @@
-require 'subjects/cellect_client'
-
-class SeenCellectWorker
+class NotifySubjectSelectorOfSeenSubjectWorker
```
Comment (camallen) [inline app/workers/notify_subject_selector_of_seen_subject_worker.rb]: the second subject seems redundant. not super keen on these names but maybe `NotifySubjectSelectorOfSeenWorker`

File: app/workers/notify_subject_selector_of_subjects_change_worker.rb [hunk 3]
```diff
@@ -1,6 +1,4 @@
-require 'subjects/cellect_client'
-
-class ReloadCellectWorker
+class NotifySubjectSelectorOfSubjectsChangeWorker
```
Comment (camallen) [inline app/workers/notify_subject_selector_of_subjects_change_worker.rb]: `NotifySubjectSelectorOfChangeWorker` ?

File: lib/subjects/panoptes_client.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+module Subjects
```
Comment (camallen) [inline lib/subjects/panoptes_client.rb]: Should this be in `app/services`?

Comment (marten) [inline lib/subjects/panoptes_client.rb]: Ah, it's no longer referenced due to refactorings. Deleting.

**Comment by marten** [comment]
I'll incorporate your comments and merge. I've got more refactorings coming, but let's get the notifications stuff out the door.

**Pull request merged**